File: leetcode-go/Array/LC0217-存在重复元素.go
package main

import (
	"fmt"
	"sort"
)

/**
 *
 * 输入：nums = [1,2,3,1]
 * 输出：true
 * 输入：nums = [1,2,3,4]
 * 输出：false
 */

// 哈希表法
func containsDuplicate(nums []int) bool {
	m := make(map[int]int)
	for _, v := range nums {
		if _, ok := m[v]; ok {
			return true
		}
		m[v]++
	}
	return false
}

// 排序法, 前后比较
func containsDuplicate2(nums []int) bool {
	sort.Ints(nums)
	for i := 0; i < len(nums)-1; i++ {
		if nums[i] == nums[i+1] {
			return true
		}
	}
	return false
}

func main() {
	nums := []int{1, 2, 3, 1}
	fmt.Println(containsDuplicate(nums)) // true

	nums2 := []int{1, 2, 3, 4}
	fmt.Println(containsDuplicate(nums2)) // false

	nums3 := []int{1, 1, 1, 3, 3, 4, 3, 2, 4, 2}
	fmt.Println(containsDuplicate(nums3)) // true
}

File: leetcode-python/Array/LC0217-存在重复元素.py
'''
lc 存在重复元素
输入: [1,2,3,1]
输出: true

标签： 数组 排序 哈希表

解题思路：
1. 数组排序
2. 集合去重
3. 字典计数
'''

class Solution:
    def containsDuplicate(self, nums) -> bool:
        return len(nums) != len(set(nums))

class Solution:
    def containsDuplicate(self, nums) -> bool:
        nums.sort()
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                return True
        return False


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        d = {}
        for v in nums:
            d[v] = d.get(v, 0) + 1
        for key in d:
            if d[key] > 1:
                return True
        return False